src/CashuWallet.ts

```diff
@@ -177,24 +168,58 @@ class CashuWallet {
 	}
 
 	async requestTokens(amount: number, hash: string): Promise<Array<Proof>> {
-		const { blindedMessages, secrets, rs } = await this.createRandomBlindedMessages(amount);
+		const { blindedMessages, secrets, rs } = this.createRandomBlindedMessages(amount);
 		const payloads = { outputs: blindedMessages };
 		const { promises } = await this.mint.mint(payloads, hash);
 		return dhke.constructProofs(promises, rs, secrets, this.keys);
 	}
 
+	private async receiveTokenEntry(tokenEntry: TokenEntry, keys: MintKeys) {
+		const proofsWithError: Array<Proof> = [];
+		const proofs: Array<Proof> = [];
+		try {
+			const amount = tokenEntry.proofs.reduce((total, curr) => total + curr.amount, 0);
+			const { payload, amount1BlindedMessages, amount2BlindedMessages } = this.createSplitPayload(
+				0,
+				amount,
+				tokenEntry.proofs
+			);
+			const { fst, snd } = await CashuMint.split(tokenEntry.mint, payload);
+			const proofs1 = dhke.constructProofs(
+				fst,
+				amount1BlindedMessages.rs,
+				amount1BlindedMessages.secrets,
+				keys
+			);
+			const proofs2 = dhke.constructProofs(
+				snd,
+				amount2BlindedMessages.rs,
+				amount2BlindedMessages.secrets,
+				keys
+			);
+			proofs.push(...proofs1, ...proofs2);
+		} catch (error) {
+			console.error(error);
+			proofsWithError.push(...tokenEntry.proofs);
+		}
+		return {
+			proofs,
```

Comment:
I'm not even sure if we need this. Can there be single proofs that fail in a token, while others succeed?

Reply:
oh nevermind.

Reply:
sorry for the confusion, i see now that you need them in the receive function to determine wether the token was faulty. all good!